paasta_tools/setup_kubernetes_job.py

```diff
@@ -266,6 +268,17 @@ def setup_kube_deployments(
                     app.kube_deployment.instance,
                     app.kube_deployment.namespace,
                 ) not in existing_apps:
+                    if app.soa_config.get_bounce_method() == "downthenup":
+                        exist_on_other_namespace = False
+                        for existing_app in existing_apps:
+                            if existing_app[:2] == (
+                                app.kube_deployment.service,
+                                app.kube_deployment.instance,
+                            ):
+                                exist_on_other_namespace = True
+                                break
+                        if exist_on_other_namespace:
+                            continue
```

Comment:
maybe this part of the code would be easier to read as:

```
if app.soa_config.get_bounce_method() == "downthenup":
    if any((app[:2] == (app.kube_deployment.service, app.kube_deployment.instance)) for app in existing_apps):
        # For downthenup, we don't want to create until cleanup_kubernetes_job has cleaned up the instance in the other namespace.
        continue
```

Reply:
:TY: black this went from 3 lines to a bunch......


```python
if app.soa_config.get_bounce_method() == "downthenup":
    if any(
        (
            app[:2]
            == (
                app.kube_deployment.service,
                app.kube_deployment.instance,
            )
        )
        for app in existing_apps
    ):
        # For downthenup, we don't want to create until cleanup_kubernetes_job has cleaned up the instance in the other namespace.
        continue
```